src/components/Photo/Selfie.js

```diff
@@ -11,27 +11,108 @@ import style from './style.css'
+    base64: string
+  }>,
 }
 
 type Props = {
   onCapture: Function,
   renderFallback: Function,
   trackScreen: Function,
+  useMultipleSelfieCapture: boolean,
+  snapshotInterval: number,
 }
 
 const inactiveError = { name: 'CAMERA_INACTIVE', type: 'warning' }
 
 export default class Selfie extends Component<Props, State> {
   webcam = null
+  snapshotIntervalRef: ?IntervalID = null
 
   state: State = {
     hasBecomeInactive: false,
+    snapshotBuffer: [],
   }
 
   handleTimeout = () => this.setState({ hasBecomeInactive: true })
 
-  handleClick = () => screenshot(this.webcam,
-    (blob, base64) => this.props.onCapture({ blob, base64 }))
+  /* When taking snapshots we want to ensure that the snapshot we provide is, where possible,
+    older than the interval. As we take a snapshot every interval, that means we need to
+    store the previous snapshot so that we can use that if the capture is taken too soon
+    after the latest snapshot (e.g. 500ms after the last snapshot on a 1000ms interval).
+    
+    Therefore we use a simple two item collection as our buffer:
+    
+    ['ready snapshot', 'fresh snapshot']
+
+    The first item is the older of the two snapshots, and will be used when capturing the selfie.
+    The second item is the 'fresh' slot which will not be suitable until the next interval tick at which
+    point it is moved to the 'ready' slot. If there are no items in the buffer, then
+    the new snapshot will be placed in both positions in the buffer, so that we always provide a
+    snapshot (even if that could sometimes be a suboptimal time delta) */
+  takeSnapshot = () => {
+    screenshot(this.webcam, (blob, base64) => {
+      this.setState(({ snapshotBuffer: [, newestSnapshot] }) => ({
+        snapshotBuffer: [
+          ...([newestSnapshot] || [
+            {
+              blob: new File([blob], "applicant_snapshot.png", {
+                type: "image/png"
+              }),
+              base64
+            }
+          ] ||
+            []),
+          {
+            blob: new File([blob], "applicant_snapshot.png", {
+              type: "image/png"
+            }),
+            base64
+          }
+        ]
+      }))
+    })
+  }
+
+  handleClick = () => {
+    screenshot(this.webcam, (blob, base64) =>
+      this.props.onCapture(
+        this.props.useMultipleSelfieCapture
+          ? {
+            selfie: true,
+            snapshot: this.state.snapshotBuffer[0],
+            capture: {
+              blob: new File([blob], "applicant_selfie.png", {
+                type: "image/png"
+              }),
+              base64
+            }
+          }
+          : {
+            blob: new File([blob], "applicant_selfie.png", {
+              type: "image/png"
+            }),
+            base64
+          }
+      )
+    )
+  }
+
+  componentDidMount() {
+    setTimeout(this.takeSnapshot, this.props.snapshotInterval / 2)
+    this.snapshotIntervalRef = setInterval(
+      this.takeSnapshot,
```

Comment:
Is this working? I think this will fail if the webcam stream is not yet available.

Reply:
It will indeed fail, and it's annoying as there's no good way to listen for the webcam starting up! (or at least no way that's exposed and I could find in my admittedly short investigation). This is why the initial snapshot is delayed by half the usual snapshot delay. It's not ideal, but it avoids the issue. For reference, though, when the webcam stream is not present, it only prevents that specific snapshot being taken - it doesn't take out the snapshot functionality.

Reply:
There is a `onUserMedia` callback on the Camera component that can be used to find out if the stream is available. Also the Camera component reference is assigned to `this.webcam`

Reply:
aha! So it could just call it from that! awesome :) will update :+1:

Reply:
Note that I added this back in [the timeout] after the conversation as I was still seeing the error with this approach. Have you not been able to replicate it?